dashboard/agent.py

```diff
@@ -225,13 +221,18 @@ def callback(msg):
                 self.log_dir, self.gcs_address, callback, loop
             )
             tasks.append(check_parent_task)
-        await asyncio.gather(*tasks)
 
         if self.server:
-            await self.server.wait_for_termination()
+            tasks.append(self.server.wait_for_termination())
```

Comment:
I manually test it with a broken grpcio version but I don't know how to write a test for it since I don't know how to let `wait_for_termination()` throw an exception with a good grpcio version.